src/containers/ColorByMenu/index.tsx

```diff
@@ -184,30 +188,76 @@ class ColorByMenu extends React.Component<ColorByMenuProps> {
             downloadConfig,
             downloadRoot,
             colorBy,
+            groupBy,
             colorByMenuOptions,
+            groupByDisplayOptions,
             handleChangeAxis,
+            changeGroupByCategory,
             colorForPlot,
             categoryCounts,
             categoricalFeatures,
         } = this.props;
+
+        const showColorLegend = includes(categoricalFeatures, colorBy) && colorBy !== groupBy;
         return (
             <React.Fragment>
                 <Row className={styles.colorByRow}>
-                    <Col span={6}>Color by:</Col>
-                    <Col span={18}>
-                        <AxisDropDown
-                            axisId={COLOR_BY_SELECTOR}
+                    <Col span={4}>Color by:</Col>
+                    <Col span={20}>
+                        <FeatureSelectDropdown
                             value={colorBy as string}
                             options={colorByMenuOptions}
-                            handleChangeAxis={handleChangeAxis}
+                            onChange={(v: string) => {
+                                handleChangeAxis(COLOR_BY_SELECTOR, v);
```

Comment:
Out of scope of this PR, but `handleChangeAxis` should maybe accept specific constants (or an enum) instead of strings. I noticed that, technically, any property can be set using it? (You could technically set `groupBy` with it too)